app/services/flowcommerce_spree/order_updater.rb

```diff
@@ -45,6 +45,7 @@ def finalize_order
     def complete_checkout
       upsert_data
       map_payments_to_spree
+      finalize_order
```

Comment:
Shouldn't it be removed from the webhook then?

https://github.com/mejuri-inc/flowcommerce_spree/blob/feature/tec-4958-thankyou-page-fix/lib/flowcommerce_spree/webhook_service.rb#L178

Reply:
That's a great point. I have doubts related to removing it from the hook, thought. That call in the webhook should not be reached if the order is already completed, so there is no risk of running twice the finalize_order, and can probably help us in case there is an error when getting redirected to the thank you page and for some reason the order was not properly closed.

If you think it would be a better idea to avoid automatically trying then I can remove it. What do you think?

Reply:
Hm, your catch is very good too - let it stay!

Also, just thought that we should check order's state -`finalize_order if @order.state = 'complete'`

Reply:
Makes total sense!! I will add this validation as well here !! Thanks!